### DISCUSSION TITLE
Requests from GitHub actions blocked by `madeira.gov.pt`

### DISCUSSION THREAD

**Discussion by HarryVasanth** [carlosrsabreu/devo-abastecer#20] [General]
### 🚫 All requests from GitHub actions blocked by `madeira.gov.pt`

<img width="537" alt="Screenshot 2023-01-19 at 21 23 31" src="https://user-images.githubusercontent.com/14015057/213565412-2516bedc-8feb-420c-9616-efa530bf7a2e.png">

I've tried several ways to overcome this, the only way seems to be (for the time being) is to use our own hosted runners.
This is the reason the current workflow is failing: [![⛽️ Update Gas Prices from DRETT](https://github.com/carlosrsabreu/devo-abastecer/actions/workflows/update_gas_prices.yml/badge.svg?branch=main)](https://github.com/carlosrsabreu/devo-abastecer/actions/workflows/update_gas_prices.yml)

**Comment by HarryVasanth** [comment]
Using proxy websites doesn't seem to work either:
<img width="532" alt="Screenshot 2023-01-19 at 21 54 59" src="https://user-images.githubusercontent.com/14015057/213570698-dbf92884-74c2-4230-9504-82dd55cb4d67.png">

Tried the oldest trick in the book, [Google Translate](https://translate.google.com/translate?sl=en&tl=pt&hl=en-US&u=https://www.madeira.gov.pt/drett&client=webapp) the page, and still no luck:

<img width="460" alt="Screenshot 2023-01-19 at 21 58 01" src="https://user-images.githubusercontent.com/14015057/213571168-f562dbd5-8a34-43ea-9fa6-ab4c8b4b8c3f.png">

**Reply by Dntfreitas** [reply]
Probably a residential proxy will do the work; I don't know how to setup one for free…

**Reply by HarryVasanth** [reply]
The trick is to run the proxy inside the workflow, or get the proxied URL to retrieve the data. I tried a self-hosted runner on a VPS. [**No luck**](https://github.com/carlosrsabreu/devo-abastecer/actions/runs/3963236316/jobs/6790840557#step:6:32). Hopefully this is temporary. Let's see..

**Reply by HarryVasanth** [reply]
OK, managed to make the action runner `curl` the site with a VM in Portugal. Also noticed that we can't access `https://www.madeira.gov.pt/drett` with a VPN from another country.

**Reply by Dntfreitas** [reply]
Could they eventually block that VM IP sometime in the future?

**Reply by HarryVasanth** [reply]
As far as I can see, it was unintentional. If it is intentional, then it's like boarding up your front door, so no one can come in, and you have the side windows wide open. 💁

p.s. The vm may or may not belong to RAM 😉

**Reply by Dntfreitas** [reply]
Apparently is working now :)

**Reply by HarryVasanth** [reply]
Yes, about to write to you. It was probably a mistake or badly setup blacklisting policy.
Anyway, right on time for the new update.

**Reply by HarryVasanth** [reply]
The proxy links and translation links are also up:
https://www-madeira-gov-pt.translate.goog/drett?_x_tr_sl=auto&_x_tr_tl=pt&_x_tr_hl=en&_x_tr_pto=wapp

Definitely was a mistake on their part.

**Reply by Dntfreitas** [reply]
Good! Those are the best issues, the ones that solve by themselves.
